Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/model/Tasks.java

```diff
@@ -107,4 +120,13 @@ public boolean apply(Map.Entry<Id, SyncedData<Task>> input) {
             }
         };
     }
+
+    @Override
+    public String toString() {
+        String result = "";
+        for (Id id : internalMap().keySet()) {
+            result = result + ", " + "{" + id + "; " + internalMap().get(id).syncState() + "}";
+        }
+        return result.substring(2);
```

Comment:
It'd be indeed clearer if you extract `", "` to a final string and use its length, so that intentions are explicit.